etc/profile-m-z/youtube-viewers-common.profile

```diff
@@ -19,6 +19,11 @@ include allow-perl.inc
 include allow-python2.inc
 include allow-python3.inc
 
+# These lines are needed to allow Firefox to open links
+noblacklist ${HOME}/.mozilla
+whitelist ${HOME}/.mozilla/firefox/profiles.ini
+read-only ${HOME}/.mozilla/firefox/profiles.ini
```

Comment:
```suggestion
# These lines are needed on older Firefox versions (that use Xorg for IPC) to
# allow opening links on an existing instance of Firefox (note that it still
# fails if there are no running instances)
#noblacklist ${HOME}/.mozilla
#whitelist ${HOME}/.mozilla/firefox/profiles.ini
#read-only ${HOME}/.mozilla/firefox/profiles.ini
```

Changes:

* Explain the reasoning in more details
* Leave the code commented

Leave it commented because:

* It appears brittle/undocumented (as in, may break in future versions)
* It supposedly only affects older versions of Firefox
* It looks like it could result in Firefox creating a new profile (as the
  existing profile directories would not be visible)

The default Firefox profile is arguably rather lacking in security/privacy and
may not have the desired extensions (such as an adblocker) and network
configuration (proxies, etc).

Reply:
`profiles.ini` is required by Firefox to know to which instance it should talk (the hash of the profile name in `profiles.ini` is part of the D-Bus name).

Reply:
Could you reproduce something else than this?

![Your Firefox profile cannot be loaded. It may be missing or inaccessible.](https://user-images.githubusercontent.com/41237666/189522013-bc77cada-21c4-4ac4-a475-1d40aef9860d.png)

Reply:
@rusty-snake [on Sep 11](https://github.com/netblue30/firejail/pull/5352#discussion_r967797814):

> `profiles.ini` is required by Firefox to know to which instance it should
> talk (the hash of the profile name in `profiles.ini` is part of the D-Bus
> name).

I see; I thought it was only needed for the xorg IPC.

Does it depend on access to profiles.ini because it is used to find out which
one is the default profile?

What happens if there is only a non-default-profile instance of Firefox running
(without `--no-remote`)?

@rusty-snake [on Sep 11](https://github.com/netblue30/firejail/pull/5352#discussion_r967798603):

> > It looks like it could result in Firefox creating a new profile (as the
> > existing profile directories would not be visible
> 
> Could you reproduce something else than this?
> 
> ![Your Firefox profile cannot be loaded. It may be missing or
> inaccessible.](https://user-images.githubusercontent.com/41237666/189522013-bc77cada-21c4-4ac4-a475-1d40aef9860d.png)

I have not tested this; my suggestion above was based on what I understood from
reading the following comments from @pirate486743186 (which only mention xorg
and nothing about dbus nor default profiles):

* <https://github.com/netblue30/firejail/pull/5352#issuecomment-1234217847>
* <https://github.com/netblue30/firejail/pull/5352#issuecomment-1234244758>

Reply:
> What happens if there is only a non-default-profile instance of Firefox running
(without --no-remote)?

The same as there is no firefox running.

Reply:
@rusty-snake [on Sep 11](https://github.com/netblue30/firejail/pull/5352#discussion_r967823024):

> > What happens if there is only a non-default-profile instance of Firefox
> > running (without --no-remote)?
> 
> The same as there is no firefox running.

Nice.

Updated suggestion:

```suggestion
# The lines below are needed to find the default Firefox profile name, to allow
# opening links in an existing instance of Firefox (note that it still fails if
# there isn't a Firefox instance running with the default profile; see #5352)
noblacklist ${HOME}/.mozilla
whitelist ${HOME}/.mozilla/firefox/profiles.ini
read-only ${HOME}/.mozilla/firefox/profiles.ini
```